### PR TITLE
FOGL-8563 Add ability to tune asset tracker update frequency

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: C/services/south/include/defaults.h [hunk 1]
```diff
@@ -25,6 +25,9 @@ static struct {
 			"Enable flow control by reducing the poll rate", "boolean", "false" },
 	{ "readingsPerSec",	"Reading Rate",
 			"Number of readings to generate per interval", "integer", "1" },
+	{ "assetTrackerInterval",	"Asset Tracker Update",
+			"Number of milliseconds between udpates of the asset tracker information",
```
Comment (AmandeepArora) [inline C/services/south/include/defaults.h:29]: "updates"

File: docs/tuning_fledge.rst [hunk 2]
```diff
@@ -47,6 +47,8 @@ The south services within Fledge each have a set of advanced configuration optio
 
   - *Reading Rate* - The rate at which polling occurs for this south service. This parameter only has effect if your south plugin is polled, asynchronous south services do not use this parameter. The units are defined by the setting of the *Reading Rate Per* item.
 
+  - *Asset Tracker Update* - This control how frequently the asset tracker flushes the cache of asset tracking information to the storage layer. It is a value expressed in milliseconds. The asset tracker only write updates, therefore if you have a fix set of assets flowing in a pipeline the asset tracker will only write any data the first time each asset is seen and will then perform no further writes. If you have varaiblility in your assets or asset structure the asset tracker will be more active and it becomes more useful to tune this parameter.
```
Comment (AmandeepArora) [inline docs/tuning_fledge.rst:50]: "variability"

File: docs/tuning_fledge.rst [hunk 3]
```diff
@@ -186,6 +188,8 @@ In a similar way to the south services, north services and tasks also have advan
 
   - *Data block size* - This defines the number of readings that will be sent to the north plugin for each call to the *plugin_send* entry point. This allows the performance of the north data pipeline to be adjusted, with larger blocks sizes increasing the performance, by reducing overhead, but at the cost of requiring more memory in the north service or task to buffer the data as it flows through the pipeline. Setting this value too high may cause issues for certain of the north plugins that have limitations on the number of messages they can handle within a single block.
 
+  - *Asset Tracker Update* - This control how frequently the asset tracker flushes the cache of asset tracking information to the storage layer. It is a value expressed in milliseconds. The asset tracker only write updates, therefore if you have a fix set of assets flowing in a pipeline the asset tracker will only write any data the first time each asset is seen and will then perform no further writes. If you have varaiblility in your assets or asset structure the asset tracker will be more active and it becomes more useful to tune this parameter.
```
Comment (AmandeepArora) [inline docs/tuning_fledge.rst:191]: "fixed set of"?

File: C/common/include/asset_tracking.h [hunk 4]
```diff
@@ -21,6 +21,8 @@
 #include <condition_variable>
 #include <storage_client.h>
 
+#define MIN_ASSET_TRACKER_UPDATE	500 // The minimum interval for asset tracker updates
```
Comment (AmandeepArora) [inline C/common/include/asset_tracking.h:24]: Can even 500 ms setting be considered too aggressive in some cases given that these changes go to (separate) foglamp.db and is not competing with ingestion/reading of readings?

Comment (MarkRiddoch) [reply C/common/include/asset_tracking.h:24]: This is here for compatibility reasons as the current value is 500ms

Comment (AmandeepArora) [reply C/common/include/asset_tracking.h:24]: Ok, got it. This value may be reduced even below 500 ms. 

Can you say what was the original issue that required this change?

**Review by AmandeepArora — APPROVED** [review]

File: C/common/include/asset_tracking.h (hunk 4 above)
Comment (MarkRiddoch) [reply C/common/include/asset_tracking.h:24]: It was a request from RTE as they saw the audit logger taking a lot of bandwidth from the storage service

**Pull request merged**